src/main/scala/dal/PasswordSaltDal.scala

```diff
@@ -0,0 +1,25 @@
+package dal
+
+import sqlest._
+import domain.User._
+import dal.table.UserDetailTable
+
+
+class PasswordSaltDal extends SqlestDb {
+  def getPasswordSalt(id: UserID): Option[(String,String)]  = {
+    select(UserDetailTable.password, UserDetailTable.salt)
+      .from(UserDetailTable)
+      .where(UserDetailTable.userID.toString == id.toString)
+      .fetchHeadOption
+  }
+
+  def getPasswordAndSalt(userID:UserID): Either[String,(String,String)] = {
+    val extract: Option[(String,String)] = getPasswordSalt(userID)
+    if(extract.isEmpty) Left("User ID or password incorrect")
+    else Right(extract.get)
+  }
```

Comment:
Code is fine as is. Just mentioning it as an alternate syntax, it could be rewritten using pattern match style as below. Either approach is fine.

case getPasswordSalt(userID) match {
    None => Left("User ID or password incorrect")
    Some(x) => Right(x)
}